### ISSUE TITLE
Order updates is not websocket based.

### ISSUE THREAD

**Issue by krishnavelu** [krishnavelu/alice_blue#2]
All other brokers like zerodha & upstox has their order updates as websocket based.
Meaning you don't need to poll for getting an order update using [`get_order_history()`](https://github.com/krishnavelu/alice_blue/blob/master/alice_blue/alice_blue.py#L399-L404)
If you subscribe for order update, the subscribed function will be called whenever there is an order update.

This feature is not yet implemented in alice-blue server.

**Comment by priyakumariengineer** [comment]
Hi is there any deadline for this ??

**Comment by krishnavelu** [comment]
This should be implemented from alice blue side first. Then the support for
the same can be addd in library.
Currently there is no deadline available.

**Comment by priyakumariengineer** [comment]
Hi Krishna, 
I understand. So as per current situation we need to poll order history continuously. What is rate limit to fetch order history ??

**Comment by krishnavelu** [comment]
Currently there is no rate limit implemented. But that doesn’t mean you can
poll every millisecond. By doing so you’ll force alice blue to implement
rate limit. Same thing happened in upstox.

Poll few seconds, not milliseconds.

**Comment by developersb** [comment]
Here is got an error while WebSocket is running mode check below

```
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): ant.aliceblueonline.com:443
DEBUG:urllib3.connectionpool:https://ant.aliceblueonline.com:443 "GET /api/v2/order HTTP/1.1" 200 None
ERROR:websocket:error from callback <bound method AliceBlue.__on_error_callback of <alice_blue.alice_blue.AliceBlue object at 0x7f2e7a82f7b8>>: error_callback() missing 1 required positional argument: 'error'
```

**Comment by krishnavelu** [comment]
@developersb 
Why are you posting here? Your problem is nowhere related to ‘order update based on websocket.

This error is due to a missing positional argument ‘error’ while calling a function.
I guess you missed error argument in error callback function.

**Comment by shivendrarajawat** [comment]
does alice blue provides stream for transactions (from time of request) through heartbeat, if not, is this in the list of enhancement ?

**Comment by krishnavelu** [comment]
What do you mean by stream of transactions?

**Comment by shivendrarajawat** [comment]
> What do you mean by stream of transactions?

see this example
https://developer.oanda.com/rest-live-v20/transaction-ep/#collapse_6_example_curl_1

this kind of streaming I am talking about.

**Comment by krishnavelu** [comment]
You mean order status using websocket? That’s what this issue is about. Its still not available from server side. If its available we can implement in the library.

**Comment by aatishgupta** [comment]
Hi Krishna,
How many Token we can generate from single account for respective day.
As we know token is valid for a day only.

**Comment by krishnavelu** [comment]
Token generated is valid for a day.

**Comment by aatishgupta** [comment]
> Token generated is valid for a day.

Right token is valid for the day but how many token i can create in single day.
For example if i ran 10 script then 10 token will be generated. 
then any limitation for token generation.
Is any disadvantage of multiple token generation.

**Comment by krishnavelu** [comment]
why you need 10 tokens? Use the same token for the all scripts.
You can generate only one token per account. Once you generate 2nd token the first will be invalidated.

**Comment by aatishgupta** [comment]
> why you need 10 tokens? Use the same token for the all scripts.
> You can generate only one token per account. Once you generate 2nd token the first will be invalidated.

Thanks for clarification.

**Comment by aatishgupta** [comment]
Hi Krishna,
how to place single order in multiple account is there anyway.
Please guide me.
Thanks in advance.
Thanks for your quick response.

**Comment by krishnavelu** [comment]
This library is for single user usage. Not for multiple.

**Comment by bhaveshagrawal1014** [comment]
@krishnavelu Any update in this ticket? Can we get order updates via socket?

**Comment by krishnavelu** [comment]
I have no updates. Also I’m not following up. You can contact alice directly.

**Comment by viswanath514** [comment]
Hi Krishnavelu, 
Any update here? I mean is there websocket support for order update now? 

Regards, 
Viswanath

**Comment by krishnavelu** [comment]
No updates.

**Comment by viswanath514** [comment]
Thanks for the update.

**Comment by ashish1497** [comment]
Any update on this? Is there a new version coming?

**Comment by krishnavelu** [comment]
I need to work on the V2, Its ready from Alice side.

**Comment by jaydave1988** [comment]
Is there any update Krishnavelu?

**Comment by krishnavelu** [comment]
I was working. Left 1/2 way due to some problem. May be in april/may

**Comment by jaydave1988** [comment]
Do you have document about this? I am not able to find that in aliceblue's site.
If yes could you please share?

**Comment by krishnavelu** [comment]
https://github.com/krishnavelu/alice_blue/issues/373

**Comment by jaydave1988** [comment]
I went through the documents from #373 but did not find anything about websocket for order_update json structure.

**Comment by krishnavelu** [comment]
Yes, still documentation is not clear about websocket. I’m interacting with alice about that.

**Comment by raja-vignesh** [comment]
The developer - Tradelab itself has released an official library but it has a websocket issue and the community is not active. 
https://github.com/TradelabTechnologiesIndia/pyoauthbridge

**Comment by aeron7** [comment]
> The developer - Tradelab itself has released an official library but it has a websocket issue and the community is not active.

This is the flask library brokers like sasonline and alice used to give pre this python wrapper. https://github.com/TradelabTechnologiesIndia/pyoauthbridge/blob/master/pyoauthbridge/wsclient.py

If you see this part, it will match with sasonline's system. they have not updated since then. 
While zebull and alice, Philip capital and few uses same webapp.

**Comment by shampavman** [comment]
is the order update issue fixed yet?

**Comment by krishnavelu** [comment]
not fixed yet.

**Comment by Jothi-Vikranth** [comment]
Whether this feature is implemented ?

**Comment by krishnavelu** [comment]
No

**Comment by sagard0508** [comment]
Hello whether order update via web socket is available now or still pending?

**Comment by krishnavelu** [comment]
No.

**Comment by sagard0508** [comment]
Hi @krishnavelu , please refer https://v2api.aliceblueonline.com/websocket . here websocket of live order feed is available. Is it possible to update the similar things in your library? If there is any specific issues, then us know. Thanks

**Comment by krishnavelu** [comment]
I may update, It would take time.
